src/elm/type.js

```diff
@@ -118,9 +119,15 @@ class ToDecimal extends Expression {
   exec(ctx) {
     const arg = this.execArgs(ctx);
     if (arg != null) {
-      const decimal = limitDecimalPrecision(parseFloat(arg.toString()));
-      if (isValidDecimal(decimal)) {
-        return decimal;
+      if (arg.isUncertainty) {
```

Comment:
I don't know how I feel about this change. It was required in order to get division to work correctly. This is because unlike the other basic arithmetic operators, the arguments of the division operator are processed as though they need to go through `ToDecimal`.
Technically all is working now since I've handled the case of an uncertainty here, but it feels weird to have an uncertainty check in only this one operation. Do you feel like it needs to be added in any of the other type operations?

Reply:
This is an interesting one.  Looking at the spec, `ToDecimal` only accepts a `Boolean` or a `String`, neither of which can be expressed as an uncertainty. But I see that the test cases you created cause the CQL-to-ELM translator to automatically insert a `ToDecimal` call, passing in an uncertainty -- so it seems the translator may invoke things technically not in the spec!

I guess we probably should support something like this for some of the other `ToXXX` functions, particularly those that _can_ take an argument that can be represented by an Uncertainty.  That would be:
```
ToBoolean(argument Decimal) Boolean
ToBoolean(argument Integer) Boolean
ToDate(argument DateTime) Date
ToDateTime(argument Date) DateTime
ToQuantity(argument Decimal) Quantity
ToQuantity(argument Integer) Quantity
```

I think I would also add `ToInteger`, because I think the translator maybe would use something like that to try to convert a Decimal to an Integer.

I left `toString` off the list because the spec does not define how to translate and uncertainty to a String and it also says that the string must be "round-trippable" back to its original type.

All the said, I think it's outside the scope of this task.  So I'd suggest we create another bug to track it.